pkg/apis/core/types.go

```diff
@@ -2662,6 +2673,10 @@ type PodSpec struct {
 	// The higher the value, the higher the priority.
 	// +optional
 	Priority *int32
+	// PreemptionPolicy is the Policy for preempting pods with lower priority.
+	// This field is alpha-level and is only honored by servers that enable the NonPreemptingPriority feature.
+	// +optional
+	PreemptionPolicy apiv1.PreemptionPolicy
```

Comment:
optional alpha fields need to be a pointer to avoid serializing a value in proto format

Reply:
add validation to ValidatePodSpec that ensures that if a non-nil value is specified, it is a valid value

Reply:
Done